feat: add endpoints sendChatMessage and getChatMessages

derpierre65 opened the pull request.

porisius commented:
Okay, I love the idea, however, I see a security problem.

Anyone with access to the server could send a chat command that, depending on the other mods installed or a twitch integration, could create chaos.

I would like to see some kind of authentication mechanism to help protect. If the mod user creates their own security issue by bypassing the security, then that will be their problem not ours.

derpierre65 commented:
> Okay, I love the idea, however, I see a security problem.
> 
> Anyone with access to the server could send a chat command that, depending on the other mods installed or a twitch integration, could create chaos.
> 
> I would like to see some kind of authentication mechanism to help protect. If the mod user creates their own security issue by bypassing the security, then that will be their problem not ours.

good point, i added a very simple authorization system
i also refactored the RegisterEndpoint stuff, it's now more readable than the all boolean parameters and it's much easier to add new parameters

i am unsure about saving the configuration after generating an authentication token. it isn't working yet

i added the authorization for setSwitches and sendChatMessage
you need to set the token to the Authorization header and you can customize the token in the WebServer.cfg
just to prevent others from call write endpoints without permission
